### DISCUSSION TITLE
8.2.0 Auto Greyscale Detection

### DISCUSSION THREAD

**Discussion by Junko-Ganan** [dlemstra/Magick.NET#1058] [Help] [answered]
Hello, i recently updated my imagemagick version from 7.24 to 8.3.3 but noticed that 8.2.0 contains grayscale updates that make our black borders completely transparent and white completely with white borders on CompositeOperator.Over . I believe this is due to the auto greyscale detection. How would i turn this off?

![Servant_0300f8e2-5d93-4579-a82c-c9bef23c3f5c](https://user-images.githubusercontent.com/26074413/138063232-bc74f5b6-6133-4b2c-acf6-9f696a5ea8c7.png)
![Servant_1b4844ab-de22-43c9-aab1-3e50062ceb67](https://user-images.githubusercontent.com/26074413/138063243-dbd79cfe-b1aa-46ab-9f1d-2391fc56ef56.png)

**Comment by dlemstra** [comment]
Do you have a code sample that can be used to reproduce the issue?

**Reply by Junko-Ganan** [reply]
[Example.txt](https://github.com/dlemstra/Magick.NET/files/7380060/Example.txt)

Line:83 is where we write borderstream to imageinstance

Here is the current code being ran. We would have to put together a simpler version that can reproduce the issue for you. 🤔

Discord: Junko#6276 - include this as well in case you would like to see it on screenshare.

**Comment by ShadowsAreBlack** [comment]
The code above has full gif generation

![Servant_b8d03a71-c750-44d5-b9a2-ca9306803255](https://user-images.githubusercontent.com/63431248/138069787-3225c2c6-17ea-4867-90f4-6adad4e81ae6.gif)
![Servant_b0e4c66a-41f5-4ef2-8008-a2da01b7f907](https://user-images.githubusercontent.com/63431248/138069795-f6ea97c3-0d9f-486d-9f03-92edd5d30f9a.gif)
![Servant_1857f67f-94b0-444e-a662-3a1ff2c97a1e](https://user-images.githubusercontent.com/63431248/138069793-b8bd8862-4455-41a5-a7fa-fb1ede8a5e71.gif)

Q8 8.2+ produces the white screen

Q16 8.2+ produces a completely transparent version

and the one where the border is visible can be replicated with anything below Q16 8.2 which is the desired result.

All using the exact same code

Borders with colors actually work, this only happens to the borders that I have force inverted in photoshop and the ones where I set the saturation to 0 to get a B&W effect

**Comment by dlemstra** [comment]
Is it possible to create a smaller sample that demonstrates this issue?

**Reply by Junko-Ganan** [reply]
[ExampleCode.txt](https://github.com/dlemstra/Magick.NET/files/7385603/ExampleCode.txt)

Here is a quick example of the code, i just placed imgur images and a write location variable for you to fill out to your specifics. Since our images are hosted locally on filesystem.

**Comment by dlemstra** [comment, marked as answer]
It looks like this is caused by setting the `ColorSpace` of the `QuantizeSettings`. You don't need to set this, and if you remove this setting your issue seems to be resolved.

**Reply by Junko-Ganan** [reply]
This did solve it, thank you for the help! 
![Servant_84c5d6ea-d38a-4772-96a8-f6a4af1726e8](https://user-images.githubusercontent.com/26074413/138603131-1bd75831-92c2-4651-94ba-22ee97321991.gif)
